src/App.tsx

```diff
@@ -1,39 +1,66 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+    timerId: 0,
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount(): void {
+    const timerId = window.setInterval(() => {
+      const name = getRandomName();
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+      if (this.state.hasClock) {
+        console.debug(`Renamed from ${this.state.clockName} to ${name}`);
+      }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+      this.setState({ clockName: name });
+    }, 3300);
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    this.setState({ timerId });
+    document.addEventListener('contextmenu', this.handleContextMenu);
 
-        {' time is '}
+    document.addEventListener('click', this.handleMouseClick);
+  }
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+  componentWillUnmount() {
+    window.clearInterval(this.state.timerId);
+    document.removeEventListener('contextmenu', this.handleContextMenu);
+    document.removeEventListener('click', this.handleMouseClick);
+  }
+
+  handleMouseClick = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: true });
+  };
+
+  handleContextMenu = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {this.state.hasClock && (
+          <Clock
+            clockName={this.state.clockName}
+            hasClock={this.state.hasClock}
```

Comment:
The 'hasClock' prop is unnecessary in the 'Clock' component as the 'Clock' will only be rendered when 'hasClock' is true.